corehq/apps/hqmedia/models.py

```diff
@@ -685,24 +685,33 @@ def create_mapping(self, multimedia, form_path, save=True):
                 updated_doc = self.get(self._id)
                 updated_doc.create_mapping(multimedia, form_path)
 
-    def get_media_objects(self):
+    def get_media_objects(self, languages=None):
         """
             Gets all the media objects stored in the multimedia map.
+            If passed a profile, will only get those that are used
+            in a language in the profile.
         """
         found_missing_mm = False
+        filter_multimedia = languages and self.media_language_map
+        if filter_multimedia:
+            media_list = []
+            for lang in languages:
+                media_list += self.media_language_map[lang].media_refs
+            requested_media = set(media_list)
         # preload all the docs to avoid excessive couch queries.
         # these will all be needed in memory anyway so this is ok.
         expected_ids = [map_item.multimedia_id for map_item in self.multimedia_map.values()]
```

Comment:
Seems like this should be filtered to only load docs with paths in `requested_media`. Something like
```py
expected_ids = [item.multimedia_id
    for path, item in self.multimedia_map.iteritems()
    if path in requested_media]
```